sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/interpreter/operator/arithmetic/BeamSqlArithmeticExpression.java

```diff
@@ -53,10 +54,8 @@ protected BeamSqlArithmeticExpression(List<BeamSqlExpression> operands, SqlTypeN
 
   @Override public BeamSqlPrimitive<? extends Number> evaluate(Row inputRow,
       BoundedWindow window) {
-    BigDecimal left = BigDecimal.valueOf(
-        Double.valueOf(opValueEvaluated(0, inputRow, window).toString()));
-    BigDecimal right = BigDecimal.valueOf(
-        Double.valueOf(opValueEvaluated(1, inputRow, window).toString()));
+    BigDecimal left = SqlFunctions.toBigDecimal((Number) opValueEvaluated(0, inputRow, window));
```

Comment:
Why not use the [`toBigDecimal(Object)`](https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java#L1673) overload

Reply:
Because the compiler error didn't say that was an option. I'll change it.